sequencer/src/l1_client.rs

```diff
@@ -121,26 +121,32 @@ impl L1Client {
 
     /// Get information about the given block.
     ///
-    /// If the desired block number is not available yet, this function will block until it becomes
-    /// available.
-    pub async fn wait_for_block(&self, number: u64) -> L1BlockInfo {
+    /// If the desired block number is not finalized yet, this function will block until it becomes
+    /// finalized.
+    pub async fn wait_for_finalized_block(&self, number: u64) -> L1BlockInfo {
         let interval = self.provider.get_interval();
 
-        // Wait for the block to become available.
-        loop {
-            match self.provider.get_block_number().await {
-                Ok(height) if height > number.into() => break,
-                Ok(height) => {
-                    tracing::warn!(number, %height, "waiting for L1 block");
-                }
-                Err(err) => {
-                    tracing::error!(%err, "failed to get L1 block height");
-                }
+        // Wait for the block to finalize.
+        let finalized = loop {
+            let Some(block) = self.get_finalized_block().await else {
+                tracing::info!("waiting for finalized block");
+                sleep(interval).await;
+                continue;
+            };
+            if block.number >= number {
+                break block;
             }
+            tracing::info!(current_finalized = %block.number, "waiting for finalized block");
             sleep(interval).await;
+            continue;
+        };
+
+        if finalized.number == number {
+            return finalized;
         }
 
-        // Get the block, retrying until we succeed.
+        // The finalized block may have skipped over the block of interest. In this case, our block
```

Comment:
I'm not 100% sure if we have adopted a standard on this, but comments are consistently rolling over to the next line. I *feel* they should be 80 chars, but maybe we don't care about this :shrug:

Reply:
I think Rust code tends to use a 100 character line length. At least that's what rustfmt does for code lines by default

Reply:
Ok my mistake